distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -88,10 +88,20 @@ type Table
                         Column.from_vector (v.at 0) (v.at 1) . java_column
                 Column.Value java_col -> java_col
                 _ -> invalid_input_shape
-        if cols.is_empty then Error.throw (Illegal_Argument.Error "Cannot create a table with no columns.") else
-            if (cols.all c-> c.getSize == cols.first.getSize).not then Error.throw (Illegal_Argument.Error "All columns must have the same row count.") else
-                if cols.distinct .getName . length != cols.length then Error.throw (Illegal_Argument.Error "Column names must be distinct.") else
-                    Table.Value (Java_Table.new cols)
+        Panic.recover Illegal_Argument <|
+            if cols.is_empty then
+                Panic.throw (Illegal_Argument.Error "Cannot create a table with no columns.")
+
+            if cols.distinct .getName . length != cols.length then
+                Panic.throw (Illegal_Argument.Error "Column names must be distinct.")
+
+            mismatched_size_column = cols.find if_missing=Nothing c->
```

Comment:
Since there aren't going to be a lot of columns most of the time, I think it would be good to report here the lengths of all the columns so it's easier for the user to know which columns are wrong.

Reply:
IMO if we _just_ list lengths of all columns it will be harder to find which ones fit and which ones do not, especially if there are >5 columns which is rather common.

IMO best to just show one that is wrong - then the user can fix it - if there are any more wrong ones - a new one will appear.

Reply:
Not just the lengths, but the names of the columns as well, so it wouldn't be confusing. If we only show one, and there are 10 columns with different lengths, that would take 9-10 runs to find all the problems.

Reply:
Yeah, but keep in mind that our IDE error message box is currently very small, so the bigger the error is the less readable it becomes.

I'm happy to revisit it at later point, e.g. when our error messages will be displayed larger. For now IMO this is already improving that we know at least one 'offending example', so I'd keep it as is.

---

If the user has just a few columns they will just see the sizes. If there's more of them, the message will be too long to easily see anyway - the user can always do `columns.map .row_count` or something like that.